File: ZMQ/lec-05-prg-03-pub-sub-basic-server.py
#
#   Weather update server
#   Binds PUB socket to tcp://*:5556
#   Publishes random weather updates
#
#   Reference: https://zguide.zeromq.org/docs/chapter1/#Getting-the-Message-Out

import zmq
from random import randrange

# Added from the original code
print("Publishing updates at weather server...") 

context = zmq.Context()
socket = context.socket(zmq.PUB)
socket.bind("tcp://*:5556")

while True:
    zipcode = randrange(1, 100000)
    temperature = randrange(-80, 135)
    relhumidity = randrange(10, 60)

    socket.send_string(f"{zipcode} {temperature} {relhumidity}")

File: ZMQ/lec-05-prg-03-pub-sub-basic-server.js
const zmq = require("zeromq");

console.log("Publishing updates at weather server...");

async function run() {
  const sock = new zmq.Publisher();

  await sock.bind("tcp://*:5556");

  while (true) {
    const zipcode = Math.floor(Math.random() * 5) + 10000;
    const temperature = Math.floor(Math.random() * (135 + 80)) - 80;
    const relhumidity = Math.floor(Math.random() * 60) + 10;

    await sock.send([zipcode, temperature, relhumidity]);
    await new Promise((resolve) => setTimeout(resolve, 500));
  }
}

run();
